### PR TITLE
fix: [M3-7623] - Fix VPC subnet Linode assignment text input

### PR DESCRIPTION
**Pull request by jdamore-linode**
## Description 📝
Fixes an issue in the "Assign Linodes to Subnet" drawer where input is not accepted when the user types into the "Linodes" field. Additionally removes some failing and obsolete tests.

## Changes  🔄
- Fix Linode assignment drawer "Linodes" field by replacing `<Autocomplete />` with `<LinodeSelect />`
- Fix Linode assignment drawer "Configuration profile" field by removing `inputValue` prop
- Update test in `vpc-linodes-update.spec.ts` to assert that text field input is accepted
- Remove disabled feature flag state tests in `vpc-navigation.spec.ts` and `vpc-landing-page.spec.ts` since the API is now responding with the VPC capability enabled

### Confirm test changes:
Run `yarn && yarn build && yarn start:manager:ci`, and then run the tests and confirm they all pass:

```bash
yarn cy:run -s "cypress/e2e/core/vpc/vpc-linodes-update.spec.ts,cypress/e2e/core/vpc/vpc-navigation.spec.ts,cypress/e2e/core/vpc/vpc-landing-page.spec.ts"
```

### Confirm text field changes:

1. Navigate to a VPC at `/vpc/:id`, creating one first if necessary
2. If necessary, create a subnet
3. In the subnet's action menu, click "Assign Linodes"
4. Confirm that when you type in the "Linodes" field, your input appears in the text field and the drop-down filters the results as expected
5. Select a Linode with more than one configuration and confirm that the "Configuration profile" field appears and can also be typed into

## As an Author I have considered 🤔

*Check all that apply*

- [ ] 👀 Doing a self review
- [ ] ❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
- [ ] 🤏 Splitting feature into small PRs
- [ ] ➕ Adding a changeset
- [ ] 🧪 Providing/Improving test coverage
- [ ] 🔐 Removing all sensitive information from the code and PR description
- [ ] 🚩 Using a feature flag to protect the release
- [ ] 👣 Providing comprehensive reproduction steps
- [ ] 📑 Providing or updating our documentation
- [ ] 🕛 Scheduling a pair reviewing session
- [ ] 📱 Providing mobile support
- [ ] ♿  Providing accessibility support

### CODE REVIEW

**Review by bnussman-akamai — APPROVED** [review]
Confirmed the Autocomplete issue is fixed and tests are passing.

This is kind of unrelated to this PR's focus, but @dwiley-akamai, does it make sense to add a loading state to this button? 
Screenshot 2024-01-09 at 3 51 00 PM

**Review by dwiley-akamai — DISMISSED** [review]
Tests passing ✅
Code review ✅

Can we get a changeset entry for the Cloud change? (and optionally one for the E2E tests if you think it's needed)

**Comment by dwiley-akamai** [comment]
> This is kind of unrelated to this PR's focus, but @dwiley-akamai, does it make sense to add a loading state to this button?

Yes, I don't think it'd hurt. I do believe that part of the flow uses the raw API method as opposed to a RQ query though, so the `loading` logic will need to be written out manually

**Comment by bnussman-akamai** [comment]
Ah, understood. Let's worry about it in a follow up @dwiley-akamai

File: packages/manager/CHANGELOG.md [hunk 1]
```diff
@@ -8,7 +8,12 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/) and this p
 
 ### Fixed:
 
-- VPC docs links on VPC landing, Create, and Detail pages and in "Assign Linodes" flow
+- VPC docs links on VPC landing, Create, and Detail pages and in "Assign Linodes" flow ([#10050](https://github.com/linode/manager/pull/10050))
+- VPC subnet Linode assignment text field input issue ([#10047](https://github.com/linode/manager/pull/10047))
+
+## Tests:
```
Comment (dwiley-akamai) [inline packages/manager/CHANGELOG.md:14]:
```suggestion
### Tests:
```

**Review by dwiley-akamai — COMMENTED** [review]
Text input fix ✅
Changelog entries ✅
E2E test changes ✅

File: packages/manager/CHANGELOG.md (hunk 1 above)
Comment (jdamore-linode) [reply packages/manager/CHANGELOG.md:14]: Ah man 🥲 thanks for catching this!

**Review by dwiley-akamai — APPROVED** [review]

**Review by bnussman-akamai — APPROVED** [review]
Code changes look good!

Ran the failures a saw locally (didn't run the entire suite):
`linodes/switch-linode-state.spec.ts` passed locally
`longview/longview.spec.ts` passed locally

**Pull request merged**